Data/sync/Sync.swift

```diff
@@ -64,101 +66,137 @@ enum SyncActions: Int {
     case create = 0
     case update = 1
     case delete = 2
-
+    
 }
 
-class Sync: JSInjector {
+public class Sync: JSInjector {
     
-    static let SeedByteLength = 32
+    public static let SeedByteLength = 32
     /// Number of records that is considered a fetch limit as opposed to full data set
     static let RecordFetchAmount = 300
-    static let shared = Sync()
-
-    /// This must be public so it can be added into the view hierarchy 
-    var webView: WKWebView!
-
+    public static let shared = Sync()
+    
+    /// This must be public so it can be added into the view hierarchy
+    public var webView: WKWebView!
+    
     // Should not be accessed directly
     fileprivate var syncReadyLock = false
-    var isSyncFullyInitialized = (syncReady: Bool, fetchReady: Bool, sendRecordsReady: Bool, fetchDevicesReady: Bool, resolveRecordsReady: Bool, deleteUserReady: Bool, deleteSiteSettingsReady: Bool, deleteCategoryReady: Bool)(false, false, false, false, false, false, false, false)
+    var isSyncFullyInitialized = (syncReady: Bool,
+                                  fetchReady: Bool,
+                                  sendRecordsReady: Bool,
+                                  fetchDevicesReady: Bool,
+                                  resolveRecordsReady: Bool,
+                                  deleteUserReady: Bool,
+                                  deleteSiteSettingsReady: Bool,
+                                  deleteCategoryReady: Bool)(false, false, false, false, false, false, false, false)
     
-    var isInSyncGroup: Bool {
+    public var isInSyncGroup: Bool {
         return syncSeed != nil
     }
     
     fileprivate var fetchTimer: Timer?
-
+    
+    var baseSyncOrder: String? {
+        get {
+            return UserDefaults.standard.string(forKey: prefBaseOrder)
+        }
+        set(value) {
+            UserDefaults.standard.set(value, forKey: prefBaseOrder)
+            UserDefaults.standard.synchronize()
+        }
+    }
+    
     // TODO: Move to a better place
     fileprivate let prefNameId = "device-id-js-array"
     fileprivate let prefNameName = "sync-device-name"
     fileprivate let prefNameSeed = "seed-js-array"
     fileprivate let prefFetchTimestamp = "sync-fetch-timestamp"
+    fileprivate let prefBaseOrder = "sync-base-order"
+    
+    fileprivate lazy var isDebug: Bool = { return AppConstants.BuildChannel == .developer }()
+    
+    fileprivate lazy var serverUrl: String = {
+        return AppConstants.BuildChannel == .developer ?
```

Comment:
Use the above `isDebug`, that way logic is centralized:
`return isDebug ? "" : ""`